### PR TITLE
IJSDK-462 Navigation content for the SDK

### PR DESCRIPTION
**Pull request by JohnHake**
New content describing navigation functionality, and OpenFileDescriptor in particular.

/cc @yole  review

### CODE REVIEW

File: reference_guide/navigation/navigation_intro.md [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+interface comments for more information about these two methods.
+
+The `navigate()` method is called to open the (content-containing) object in an editor or display, and request focus. Typically an implementation
+of `Navigatable` does the heavy lifting of invoking an editor for the content and requesting focus. See below for examples,
+and the [Navigatable](https://github.com/JetBrains/intellij-community/blob/master/platform/core-api/src/com/intellij/pom/Navigatable.java)
+interface for more information.  
+
+### Example IntelliJ Platform Implementations of Navigatable
+This section describes some example implementations that can be found in the IntelliJ Platform. This is by no means an exhaustive list, 
+but is intended to illustrate how `Navigatable` works in different contexts.
```
Comment (yole) [inline reference_guide/navigation/navigation_intro.md]: `NavigatablePsiElement` and `OpenFileDescriptor` are not "example implementations", these are the main two ways plugin developers will interact with navigation functionality. This particular section should have a title "Navigating to PSI Elements" and should say that:
  * most PSI elements support navigation by implementing `NavigatablePsiElement`
  * if you have a PSI element and want to navigate to it, you can usually do that by casting it to `NavigatablePsiElement` and then invoking `navigate()`
  * if you implement a PSI element subclass, you usually want to support navigation for it, and you can do this by either ensuring that your base class implements `NavigatablePsiElement` or, if not, by implementing it yourself.

File: reference_guide/navigation/navigation_intro.md [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+#### Example Navigatable Class Related to Psi Elements
+
+The [NavigatablePsiElement](https://github.com/JetBrains/intellij-community/blob/master/platform/core-api/src/com/intellij/psi/NavigatablePsiElement.java) 
+interface is the basis for many implementation classes, abstract classes, and interfaces relating to `Navigatable` 
+Psi elements. The class diagram below shows `NavigatablePsiElement` inherits from three interfaces defining the underlying
+functionality of a `Navigatable` Psi element.
+
+<img src="img/navigatable_psi_element.png" alt="NavigatablePsiElement" width="500px"/>
+
+For example, the subclass [JavaStubPsiElement](https://github.com/JetBrains/intellij-community/blob/master/java/java-psi-impl/src/com/intellij/psi/impl/source/JavaStubPsiElement.java)
```
Comment (yole) [inline reference_guide/navigation/navigation_intro.md]: This is true but irrelevant. Plugin developers almost never need to care about `JavaStubPsiElement`.

File: reference_guide/navigation/navigation_intro.md [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+
+<img src="img/navigatable_psi_element.png" alt="NavigatablePsiElement" width="500px"/>
+
+For example, the subclass [JavaStubPsiElement](https://github.com/JetBrains/intellij-community/blob/master/java/java-psi-impl/src/com/intellij/psi/impl/source/JavaStubPsiElement.java) 
+implementation is the basis for a variety of Psi element classes that
+represent Java files, classes, methods, and fields.
+
+#### Example Navigatable Class Related to Files
+
+The [OpenFileDescriptor](https://github.com/JetBrains/intellij-community/blob/master/platform/platform-api/src/com/intellij/openapi/fileEditor/OpenFileDescriptor.java)
```
Comment (yole) [inline reference_guide/navigation/navigation_intro.md]: `OpenFileDescriptor` is also not just an example implementation, it's one of the common APIs plugin developers use to trigger navigation. It allows a plugin developer to open a specific source location in the editor, by specifying the file, line and column. It also supports more complex use cases such as controlling navigation within a split editor. This section should describe how a plugin developer can create an `OpenFileDescriptor` instance and use it to trigger navigation.

**Comment by JohnHake** [comment]
Closing temporarily.

**Pull request closed without merging**